src/arduino.cc/builder/builder_utils/utils.go

```diff
@@ -258,6 +289,24 @@ func nonEmptyString(s string) bool {
 	return s != constants.EMPTY_STRING
 }
 
+func CheckIfRecompileIsAvoidable(corePath string, archiveFile string) bool {
```

Comment:
There are some tricky details, indeed, which should be tackled in a saner way. The inter-dependencies problem is one (and could be solved by tracking also the parent core). Note that the current builder does not wipe the environment if `platform.txt` changes (because it's not part of the .d dance). Tracking dependencies via .d files is really handy for libraries or mutable code, while the core is not really mutable (in Create it's even "static"). To be even more conservative, one could track all the modification dates on all core folders and wipe if even one file is newer than the precompiled core (this could be time consuming in case of slow disks or lots of cores).